### PR TITLE
windows support

### PR DESCRIPTION
**Pull request by jerch**
Attempt to get things working under windows.

TODO:
- [x] create POSIX / Windows platform switch for:
  - [x] shell being used
  - [x] logic branching
- [x] emscripten:
  - [x] fix for `emcc` path issue in windows subshell
- [x] zig:
  - [x] better sdk folder handling (symlinks dont work)
  - [x] better `unzip` fix
- [x] wabt tools:
  - [x] always refer to current node executable
- [x] minimal globbing support for `inwasm lib/*.wasm.js` to overcome missing support in windows shell
- [x] CI tests for windows

### CODE REVIEW

**Comment by jerch** [comment]
A ranting sidenote:

While the workarounds of the first commit here were quickly hacked in, the whole process to get there took me like 5 hours, mainly waiting for nodejs to get through a task. Am I doing something wrong or is nodejs really that bad under windows? Is it too POSIX-ish internally, thus has to go through many compat layers? Or is it the poor performance of the FS again? There are at least some indications pointing in that direction:
- Any npm task stalls for 1-2s right at the beginning - module loading too stressful / FS blocking things? Note that under linux there is no perceivable waiting time.
- Unzipping of the zig sdk takes forever. The toolchain brings a stripped down llvm with many many files in a big folder tree. Funny - the MinGW `unzip.exe` port takes ~3 min to get the archive unzipped, while the windows explorer told me, that it will take >10min (did not let it complete). Note again, that this task runs in <3s under linux. WTH?
- Deleting the unzipped zig sdk takes ~30s, while linux does that in <1s. And windows even cares to show me a throughout graph when deleting from the explorer. Whats that good to for? To make the bad throughput more evident? Seriously, whats wrong with the FS under windows?

This was all done on the same machine, and while I did not really measure runtime, it feels like at least being 5 times slower. At this point I dont understand how ppl successfully develop nodejs stuff under windows - it is way too sluggish for my taste (yes I was several times at the "screw that, no one can work with that system" point).
So kudos to anyone wasting time to develop nodejs apps under windows (all those electron apps :see_no_evil:), you prolly would be several times faster to get things done under linux or macos, just because the runtime execution gets not in your way (not to speak of later user experience with the same shitty execution times). I see why WSL was needed, but why dont fix the issues in the first place?

**TL;DR**: I will finish the windows port, but dont expect me to test it thoroughly, as I cant bear its slowness.
